source/Subsystem/JobSystem.h

```diff
@@ -26,15 +19,19 @@ class JobSystem
 	~JobSystem();
 
 	bool Init();
-	bool AddJob(const std::function<bool()> f);
-	void Terminate();
+	void Update(float aFrameTime);
+	bool Terminate();
 
-	bool CollectAllThreads();
+	void AddJob(const std::function<bool()> f);
 
 private:
-	int CollectOneThread();
+	std::vector<Job> myQueuedJobs;
+	std::vector<Job> myRunningJobs;
+	std::vector<std::thread*> myThreads;
+
+	void StartJob(Job& aJob, std::thread*& aThreadOut);
+	void EndJob(Job& aJob);
 
-	std::vector<Job> myJobs;
 	const float myMaximumExpirationTime = 3.f;
```

Comment:
These member variables could be initialized in the source file instead. I understand that this should not be modified and that makes it a bit tricky, but you can keep it inaccessible and not write to it in the JobSystem.

Reply:
Oh lol that const variable isn't even used anymore, simply forgot to remove it. I believe the vectors should be kept in the header file, since someone might wonder how big the class is and what data is being managed here.